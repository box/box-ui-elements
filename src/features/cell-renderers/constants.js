// @flow

const EMPTY_VALUE = 'â€”';
const BYTES_IN_MB = 1048576;
const ANONYMOUS_USER_ID = '2';

export { EMPTY_VALUE, BYTES_IN_MB, ANONYMOUS_USER_ID };
