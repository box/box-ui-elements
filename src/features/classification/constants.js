// @flow
const DEFAULT_MAX_APP_COUNT = 3;

const SECURITY_CONTROLS_FORMAT: {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_BTN: 'shortWithBtn',
} = {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_BTN: 'shortWithBtn',
};

const ACCESS_POLICY_RESTRICTION: {
    APP: 'app',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    FTP: 'ftp',
    SHARED_LINK: 'sharedLink',
    WATERMARK: 'watermark',
} = {
    SHARED_LINK: 'sharedLink',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    APP: 'app',
    FTP: 'ftp',
    WATERMARK: 'watermark',
};

const DOWNLOAD_CONTROL: {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
} = {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
};

const LIST_ACCESS_LEVEL: {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
} = {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
};

const MANAGED_USERS_ACCESS_LEVEL: {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
} = {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
};

const SHARED_LINK_ACCESS_LEVEL: {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
} = {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
};

const APP_RESTRICTION_MESSAGE_TYPE = {
    DEFAULT: 'default',
    WITH_APP_LIST: 'withAppList',
    WITH_OVERFLOWN_APP_LIST: 'withOverflownAppList',
};

const CLASSIFICATION_COLOR_ID_0 = 0;
const CLASSIFICATION_COLOR_ID_1 = 1;
const CLASSIFICATION_COLOR_ID_2 = 2;
const CLASSIFICATION_COLOR_ID_3 = 3;
const CLASSIFICATION_COLOR_ID_4 = 4;
const CLASSIFICATION_COLOR_ID_5 = 5;
const CLASSIFICATION_COLOR_ID_6 = 6;
const CLASSIFICATION_COLOR_ID_7 = 7;

const DEFAULT_CLASSIFICATION_COLOR_ID = CLASSIFICATION_COLOR_ID_0;

export {
    ACCESS_POLICY_RESTRICTION,
    APP_RESTRICTION_MESSAGE_TYPE,
    CLASSIFICATION_COLOR_ID_0,
    CLASSIFICATION_COLOR_ID_1,
    CLASSIFICATION_COLOR_ID_2,
    CLASSIFICATION_COLOR_ID_3,
    CLASSIFICATION_COLOR_ID_4,
    CLASSIFICATION_COLOR_ID_5,
    CLASSIFICATION_COLOR_ID_6,
    CLASSIFICATION_COLOR_ID_7,
    DEFAULT_CLASSIFICATION_COLOR_ID,
    DEFAULT_MAX_APP_COUNT,
    DOWNLOAD_CONTROL,
    LIST_ACCESS_LEVEL,
    MANAGED_USERS_ACCESS_LEVEL,
    SECURITY_CONTROLS_FORMAT,
    SHARED_LINK_ACCESS_LEVEL,
};
