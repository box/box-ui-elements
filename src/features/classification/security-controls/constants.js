// @flow
const DEFAULT_MAX_APP_COUNT = 3;

const SECURITY_CONTROLS_FORMAT: {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_TOOLTIP: 'shortWithTooltip',
} = {
    FULL: 'full',
    SHORT: 'short',
    SHORT_WITH_TOOLTIP: 'shortWithTooltip',
};

const ACCESS_POLICY_RESTRICTION: {
    APP: 'app',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    FTP: 'ftp',
    SHARED_LINK: 'sharedLink',
} = {
    SHARED_LINK: 'sharedLink',
    DOWNLOAD: 'download',
    EXTERNAL_COLLAB: 'externalCollab',
    APP: 'app',
    FTP: 'ftp',
};

const DOWNLOAD_CONTROL: {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
} = {
    DESKTOP: 'desktop',
    FTP: 'ftp',
    MOBILE: 'mobile',
    WEB: 'web',
};

const LIST_ACCESS_LEVEL: {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
} = {
    BLACKLIST: 'blacklist',
    BLOCK: 'block',
    WHITELIST: 'whitelist',
};

const MANAGED_USERS_ACCESS_LEVEL: {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
} = {
    OWNERS_AND_COOWNERS: 'ownersCoOwners',
    OWNERS_COOWNERS_AND_EDITORS: 'ownersCoOwnersEditors',
};

const SHARED_LINK_ACCESS_LEVEL: {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
} = {
    COLLAB_AND_COMPANY_ONLY: 'companyAndCollabOnly',
    COLLAB_ONLY: 'collabOnly',
    PUBLIC: 'public',
};

export {
    ACCESS_POLICY_RESTRICTION,
    DEFAULT_MAX_APP_COUNT,
    DOWNLOAD_CONTROL,
    LIST_ACCESS_LEVEL,
    MANAGED_USERS_ACCESS_LEVEL,
    SECURITY_CONTROLS_FORMAT,
    SHARED_LINK_ACCESS_LEVEL,
};
