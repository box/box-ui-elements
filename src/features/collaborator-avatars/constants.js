// @flow

const COLLAB_GROUP_TYPE = 'group';
const COLLAB_USER_TYPE = 'user';
const COLLAB_PENDING_TYPE = 'pending';

export { COLLAB_GROUP_TYPE, COLLAB_USER_TYPE, COLLAB_PENDING_TYPE };
