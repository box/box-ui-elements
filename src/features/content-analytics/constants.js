// @flow
export const DOWNLOADS: 'DOWNLOADS' = 'DOWNLOADS';

export const PREVIEWS: 'PREVIEWS' = 'PREVIEWS';

export const USERS: 'USERS' = 'USERS';

export const METRIC = Object.freeze({
    DOWNLOADS,
    PREVIEWS,
    USERS,
});

const WEEK: 'week' = 'week';

const MONTH: 'month' = 'month';

const THREEMONTHS: 'threemonths' = 'threemonths';

const YEAR: 'year' = 'year';

export const PERIOD = Object.freeze({
    WEEK,
    MONTH,
    THREEMONTHS,
    YEAR,
});
