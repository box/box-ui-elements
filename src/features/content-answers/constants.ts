export const MOCK_LONG_PROMPT =
    'What is the significance of the character limit in the text area, and why is it important to ensure that users are notified when they exceed it? Additionally, what steps can be taken to prevent users from exceeding the character limit, and how can errors related to the character limit be handled in a user-friendly manner?';

export const TEXT_AREA = {
    MAX_LENGTH: 300,
};
