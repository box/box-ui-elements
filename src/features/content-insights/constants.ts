const DOWNLOADS = 'DOWNLOADS';
const PREVIEWS = 'PREVIEWS';
const USERS = 'USERS';

export const METRIC = Object.freeze({
    DOWNLOADS,
    PREVIEWS,
    USERS,
});

const WEEK = 'week';
const MONTH = 'month';
const THREEMONTHS = 'threemonths';
const YEAR = 'year';

export const PERIOD = Object.freeze({
    WEEK,
    MONTH,
    THREEMONTHS,
    YEAR,
});

const ONE_WEEK = 'one_week';
const TWO_WEEKS = 'two_weeks';
const ONE_MONTH = 'one_month';
const TWO_MONTHS = 'two_months';
const THREE_MONTHS = 'three_months';
const SIX_MONTHS = 'six_months';
const ONE_YEAR = 'one_year';
const TWO_YEARS = 'two_years';

export const PRESET_TIMES = Object.freeze({
    ONE_WEEK,
    TWO_WEEKS,
    ONE_MONTH,
    TWO_MONTHS,
    THREE_MONTHS,
    SIX_MONTHS,
    ONE_YEAR,
    TWO_YEARS,
});

export const PRESET_TO_TIMESTAMPS_MAP = {
    [PERIOD.WEEK]: {
        currentPeriod: PRESET_TIMES.ONE_WEEK,
        previousPeriod: PRESET_TIMES.TWO_WEEKS,
    },
    [PERIOD.MONTH]: {
        currentPeriod: PRESET_TIMES.ONE_MONTH,
        previousPeriod: PRESET_TIMES.TWO_MONTHS,
    },
    [PERIOD.THREEMONTHS]: {
        currentPeriod: PRESET_TIMES.THREE_MONTHS,
        previousPeriod: PRESET_TIMES.SIX_MONTHS,
    },
    [PERIOD.YEAR]: {
        currentPeriod: PRESET_TIMES.ONE_YEAR,
        previousPeriod: PRESET_TIMES.TWO_YEARS,
    },
} as const;
