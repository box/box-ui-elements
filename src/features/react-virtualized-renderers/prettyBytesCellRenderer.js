// @flow
import { BYTES_IN_MB } from '../../constants';
import baseCellRenderer from './baseCellRenderer';
import getSize from '../../utils/size';

import type { CellRendererParams } from './flowTypes';

const prettyBytesCellRenderer = (multiplier: number = BYTES_IN_MB) => (cellRendererParams: CellRendererParams) =>
    baseCellRenderer(cellRendererParams, cellValue => getSize(Number(cellValue) * multiplier));

export default prettyBytesCellRenderer;
