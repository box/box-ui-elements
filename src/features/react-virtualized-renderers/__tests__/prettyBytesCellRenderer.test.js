// @flow
import getSize from '../../../utils/size';

import { BYTES_IN_MB } from '../../../constants';

import prettyBytesCellRenderer from '../prettyBytesCellRenderer';

jest.mock('../../../utils/size');

describe('features/react-virtualized-renderers/prettyBytesCellRenderer', () => {
    let cellRendererParams;

    beforeEach(() => {
        cellRendererParams = {
            cellData: 2000,
        };
        getSize.mockImplementation(value => `${value} petaflops`);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('should render a dash when cellData is missing', () => {
        cellRendererParams.cellData = null;
        expect(prettyBytesCellRenderer()(cellRendererParams)).toBe('â€”');
    });

    test('should render a PrettyBytesCell', () => {
        expect(prettyBytesCellRenderer()(cellRendererParams)).toMatchSnapshot();
    });

    test('should set default multiplier when not provided', () => {
        cellRendererParams.cellData = 1;
        prettyBytesCellRenderer()(cellRendererParams);
        expect(getSize).toHaveBeenCalledTimes(1);
        expect(getSize).toHaveBeenCalledWith(BYTES_IN_MB);
    });

    test('should call getSize with provided multiplier', () => {
        cellRendererParams.cellData = 1;

        prettyBytesCellRenderer(200)(cellRendererParams);
        expect(getSize).toHaveBeenLastCalledWith(200);
        expect(getSize).toHaveBeenCalledTimes(1);

        prettyBytesCellRenderer(500)(cellRendererParams);
        expect(getSize).toHaveBeenLastCalledWith(500);
        expect(getSize).toHaveBeenCalledTimes(2);
    });

    test('should call getSize with numeric equivalent of cellData', () => {
        cellRendererParams.cellData = '123';

        prettyBytesCellRenderer(1)(cellRendererParams);
        expect(getSize).toHaveBeenCalledWith(123);
        expect(getSize).toHaveBeenCalledTimes(1);
    });
});
