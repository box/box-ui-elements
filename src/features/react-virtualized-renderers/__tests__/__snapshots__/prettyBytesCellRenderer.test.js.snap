// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/react-virtualized-renderers/prettyBytesCellRenderer should render a PrettyBytesCell 1`] = `"2097152000 petaflops"`;
